src/UglyToad.PdfPig/DocumentLayoutAnalysis/DecorationTextBlockClassifier.cs

```diff
@@ -0,0 +1,310 @@
+        /// <param name="pagesTextBlocks">The <see cref="TextBlock"/>s of every pages in the document. All of them are needed for the algorithm to work.</param>
+        /// <param name="similarityThreshold">Minimum similarity score to decide wether a block is labelled as decoration or not.</param>
+        /// <param name="n">Number of blocks in a page to be considered when looking for decoration blocks.</param>
+        /// <param name="maxDegreeOfParallelism">Sets the maximum number of concurrent tasks enabled. 
+        /// <para>A positive property value limits the number of concurrent operations to the set value. 
+        /// If it is -1, there is no limit on the number of concurrently running operations.</para></param>
+        public static IReadOnlyList<IReadOnlyList<TextBlock>> Get(IReadOnlyList<IReadOnlyList<TextBlock>> pagesTextBlocks,
+            double similarityThreshold = 0.25, int n = 5, int maxDegreeOfParallelism = -1)
+        {
+            return Get(pagesTextBlocks, Distances.MinimumEditDistanceNormalised, similarityThreshold, n, maxDegreeOfParallelism);
+        }
+
+        /// <summary>
+        /// Get blocks that are labelled as decoration for each page in the document, using a content and a geometric similarity measure.
+        /// <para>Decoration blocks are blocks that contains information such as author names, publication titles, page numbers, etc.
+        /// They are printed repeatedly at the border of each page, usually placed inside headers or footers, but sometimes also at the 
+        /// left or right edge of the page.</para>
+        /// </summary>
+        /// <param name="pagesTextBlocks">The <see cref="TextBlock"/>s of every pages in the document. All of them are needed for the algorithm to work.</param>
+        /// <param name="minimumEditDistanceNormalised">Minimum edit distance normalised. A value of 0 means both strings are exactly equal.</param>
+        /// <param name="similarityThreshold">Minimum similarity score to decide wether a block is labelled as decoration or not.</param>
+        /// <param name="n">Number of blocks in a page to be considered when looking for decoration blocks.</param>
+        /// <param name="maxDegreeOfParallelism">Sets the maximum number of concurrent tasks enabled. 
+        /// <para>A positive property value limits the number of concurrent operations to the set value. 
+        /// If it is -1, there is no limit on the number of concurrently running operations.</para></param>
+        public static IReadOnlyList<IReadOnlyList<TextBlock>> Get(IReadOnlyList<IReadOnlyList<TextBlock>> pagesTextBlocks,
+            Func<string, string, double> minimumEditDistanceNormalised, double similarityThreshold = 0.25, int n = 5, int maxDegreeOfParallelism = -1)
+        {
+            if (pagesTextBlocks.Count < 2)
+            {
+                throw new ArgumentException("The algorithm cannot be used with a document of less than 2 pages.", nameof(pagesTextBlocks));
```

Comment:
Might be least surprising here to return an empty list instead of throwing. While both make sense I can see a scenario where someone is automating the processing of many documents without knowledge of their length or content and they can more easily handle an empty list, which is a valid return for documents of any length, than an exception.